setup.py

```diff
@@ -47,7 +47,8 @@
     # incompatible with newer setuptools.
     RUN_REQUIRES.append('mpi4py >=3.1.1')
     SETUP_REQUIRES.append("mpi4py ==3.1.1; python_version<'3.11'")
-    SETUP_REQUIRES.append("mpi4py ==3.1.4; python_version>='3.11'")
+    SETUP_REQUIRES.append("mpi4py ==3.1.4; python_version=='3.11.*'")
+    SETUP_REQUIRES.append("mpi4py ==3.1.6; python_version>='3.12'")
```

Comment:
This listing is based on the idea that when building (with `SETUP_REQUIRES`), we want the oldest compatible version of packages, and the buiilt package will then be compatible with newer ones. I.e. if building with 3.1.5 works for Python 3.12, go for that version.

IDK if doing this is actually useful for mpi4py, though. @aragilar might have a better idea.

Reply:
I'm not familiar enough with the changes made to mpi4py to be sure (from what I can see in the changelog, the changes were mostly minor), but my feeling is locking in the setup.py wouldn't be that useful, given you probably want to install mpi4py separately to check which MPI version you're using (rather than pulling in a wheel), and use `--no-build-isolation` so you're matching the build environment to the run environment. I think leaving this as is fine for now, given mpi4py 4 looks like it's going to come this year (based on its changelog), and it's not clear if or how that will affect us.

Reply:
Thanks @aragilar ! I'll merge this as is, then.